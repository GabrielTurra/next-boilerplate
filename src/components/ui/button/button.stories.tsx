import { action } from '@storybook/addon-actions';
import type { Meta, StoryObj } from '@storybook/react';
import { Button } from './button';

export default {
  title: 'Components/Button',
  component: Button,
  tags: ['autodocs'],
  args: {
    children: 'Button',
    onClick: action('Button cliked.')
  },
  argTypes: {
    size: {
      options: ['default', 'sm', 'lg', 'icon'],
      control: 'select',
      table: {
        category: 'Appereance',
        type: {
          summary: 'enum (string)',
          detail: '"default" | "sm" | "lg" | "icon"'
        },
        defaultValue: {
          summary: 'default'
        }
      }
    },
    variant: {
      options: ['default', 'secondary', 'destructive', 'outline', 'ghost', 'link'],
      control: 'select',
      table: {
        category: 'Appereance',
        type: {
          summary: 'enum (string)',
          detail: '"default" | "secondary" | "destructive" | "outline" | "ghost" | "link"'
        },
        defaultValue: {
          summary: 'default'
        }
      }
    },
    disabled: {
      control: 'boolean',
      table: {
        category: 'Behaviour',
        type: {
          summary: 'boolean'
        },
        defaultValue: {
          summary: 'false'
        }
      }
    },
    asChild: {
      table: {
        category: 'Behaviour',
        type: {
          summary: 'boolean'
        }
      }
    },
    onClick: {
      table: {
        category: 'Event Listeners'
      }
    }
  }
} as Meta<typeof Button>;

export const Default: StoryObj<typeof Button> = {
  args: {
    asChild: false,
    size: 'default',
    variant: 'default',
    disabled: false
  }
};
