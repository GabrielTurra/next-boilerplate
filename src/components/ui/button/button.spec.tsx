import { render, screen } from '@testing-library/react';

import { Button } from './button';

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = render(<Button title="Button" />);

    expect(screen.getByRole('button', { name: /Button/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
