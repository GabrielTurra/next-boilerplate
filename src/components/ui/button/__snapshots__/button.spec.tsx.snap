// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render the button 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 focus-visible:ring-4 focus-visible:outline-1 aria-invalid:focus-visible:ring-0 bg-primary text-primary-foreground shadow-sm hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
  title="Button"
/>
`;
